Carting/src/WebApi/Infrastructure/Consumers/ProductPriceChangedEventConsumer.cs

```diff
@@ -1,24 +1,40 @@
 ﻿namespace Carting.WebApi.Infrastructure.Consumers;
 
+using System.Diagnostics;
 using Carting.WebApi.Application.Common.Interfaces;
 using Carting.WebApi.Domain.Entities;
 using MassTransit;
 using Messaging.Contracts;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
 
 internal sealed class ProductPriceChangedIntegrationEventConsumer :
     IConsumer<ProductPriceChangedIntegrationEvent>
 {
     private readonly ICartingDbContext _context;
     private readonly ILogger<ProductPriceChangedIntegrationEventConsumer> _logger;
+    private readonly TelemetryConfiguration _telemetryConfiguration;
 
-    public ProductPriceChangedIntegrationEventConsumer(ICartingDbContext context, ILogger<ProductPriceChangedIntegrationEventConsumer> logger)
+    public ProductPriceChangedIntegrationEventConsumer(ICartingDbContext context, ILogger<ProductPriceChangedIntegrationEventConsumer> logger, IConfiguration configuration)
```

Comment:
I would replace `IConfiguration configuration` with `Options` [pattern](https://docs.microsoft.com/en-us/dotnet/core/extensions/options) or just simple `TelemetryConfig` class. As this would provide instantly typed settings, also while using `Options` pattern you can take advantage of hot reload of settings if required.